lib/Controller/RoomController.php

```diff
@@ -1417,21 +1427,36 @@ public function removeAttendeeFromRoom(int $attendeeId): DataResponse {
 	/**
 	 * Allowed guests to join conversation
 	 *
-	 * @return DataResponse<Http::STATUS_OK, array<empty>, array{}>|DataResponse<Http::STATUS_BAD_REQUEST, array{error: 'breakout-room'|'type'|'value'}, array{}>
+	 * Required capability: `conversation-creation-password`
+	 *
+	 * @param string $password New password
+	 * @return DataResponse<Http::STATUS_OK, array<empty>, array{}>|DataResponse<Http::STATUS_BAD_REQUEST, array{error: 'breakout-room'|'type'|'value'|'password', hint?: null|string}, array{}>
 	 *
 	 * 200: Allowed guests successfully
 	 * 400: Allowing guests is not possible
 	 */
 	#[NoAdminRequired]
 	#[RequireLoggedInModeratorParticipant]
-	public function makePublic(): DataResponse {
+	public function makePublic(string $password = ''): DataResponse {
+		if (!$this->talkConfig->isPasswordEnforced()) {
+			try {
+				$this->roomService->setType($this->room, Room::TYPE_PUBLIC);
```

Comment:
when not empty we should still set the password with a `$this->roomService->setPassword(` call?